### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by Razan833**

### CODE REVIEW

**Comment by ashishjv1** [comment]
Please replace the Forward function in line 195 with the following code :
def forward(self, x, last=False, freeze=False):
        if freeze:
            with torch.no_grad():
                x = self.stem_conv(x)
                x = self.blocks(x)
                x = self.head_conv(x)
                #x = self.avgpool(x)
                #x = x.view(x.size(0), -1)
                x = torch.mean(x, (2, 3))
                e = self.dropout(x)
                # x = self.classifier(x)
        else:
            x = self.stem_conv(x)
            x = self.blocks(x)
            x = self.head_conv(x)
            #x = self.avgpool(x)
            #x = x.view(x.size(0), -1)
            x = torch.mean(x, (2, 3))
            e = self.dropout(x)
        out = self.classifier(x)
        if last:
            return out, e
        else:
            return out

**Comment by ashishjv1** [comment]
Make sure you don't commit directly to Main branch.
